Assets/Scripts/OrchestraArmy/Room/RoomManager.cs

```diff
@@ -0,0 +1,382 @@
+                            }
+
+                            break;
+                        case GridSpace.Wall:
+                            Spawn(x, y,
+                                BorderOnFill(x, y)
+                                    ? RoomPrefabData.WallObj
+                                    : RoomPrefabData.RockObj.transform.GetChild(Random.Range(0, 4)).gameObject);
+                            break;
+
+                        case GridSpace.DoorU:
+                            Spawn(x, y, RoomPrefabData.DoorUpObj);
+                            break;
+
+                        case GridSpace.DoorD:
+                            Spawn(x, y, RoomPrefabData.DoorDownObj);
+                            break;
+
+                        case GridSpace.DoorL:
+                            Spawn(x, y, RoomPrefabData.DoorLeftObj);
+                            break;
+
+                        case GridSpace.DoorR:
+                            Spawn(x, y, RoomPrefabData.DoorRightObj);
+                            break;
+                    }
+                }
+            }
+
+            // Spawn enemies
+            int newestEnemy = Math.Min(
+                CollectedInstrumentCount,
+                RoomPrefabData.Enemies.Count - 1
+                );
+
+            int enemyTypes = newestEnemy + 1;
+
+            int newestEnemyPercentage = (int) (100.0 / enemyTypes + 10.0);
+            int newestEnemyAmount =
+                (int) ((float) CurrentRoom.EnemySpawnLocations.Count / 100.0 * (float) newestEnemyPercentage);
+            
+            // Make sure there is always at least one new enemy in the field
+            newestEnemyAmount =
+                Math.Max(newestEnemyAmount, 1); 
+
+            foreach (Vector2 enemy in CurrentRoom.EnemySpawnLocations)
+            {
+                if (newestEnemyAmount > 0)
+                {
+                    Spawn(enemy.x, enemy.y, RoomPrefabData.Enemies[newestEnemy].gameObject);
+                    newestEnemyAmount--;
+                }
+                else
+                {
+                    // Check how many types of enemies may spawn and randomly get enemy to spawn from older ones
+                    Spawn(enemy.x, enemy.y, RoomPrefabData.Enemies[Random.Range(0, newestEnemy)].gameObject);
+                }
+            }
+
+            if (_player != null)
+            {
+                _player.transform.position = new Vector3(75 - CurrentRoom.OffsetOfRoom.x, _player.transform.position.y,
+                    75 - CurrentRoom.OffsetOfRoom.y);
+            }
+                
+        }
+
+        /// <summary>
+        /// Check if tile directly borders an empty tile
+        /// </summary>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        private bool BorderOnFill(int x, int y)
+        {
+            bool outp = false;
+            try
```

Comment:
Nu we toch aan het refactoren zijn: Kunnen hier ook wel 1 try catch van maken, catch returned toch uit de functie